smoke-test/src/test/java/org/opennms/smoketest/OpenNMSCloudPluginIT.java

```diff
@@ -0,0 +1,161 @@
+                final CreateContainerCmd createCmd = (CreateContainerCmd) cmd;
+                createCmd.withEntrypoint("/bin/bash");
+                createCmd.withName("cloudMock");
+            }).withCopyFileToContainer(MountableFile.forHostPath(PATH_TO_CLOUD_MOCK), format("%s%s", CONTAINER_PATH, CLOUD_MOCK_JAR_NAME))
+            .withCommand("-c", format("java -cp %s%s %s", CONTAINER_PATH, CLOUD_MOCK_JAR_NAME, CLOUD_MOCK_MAIN)).withExposedPorts(9003)
+            .withNetwork(stack.opennms().getNetwork())
+            .withNetworkAliases("cloudMock");
+
+    @Before
+    public void setUp() {
+        mockCloudContainer.start();
+    }
+
+    @Test
+    public void installCloudPlugin() throws Exception {
+        installCloudPluginInOpennmsMustBeSuccessful();
+        installCloudPluginInSentinelMustBeSuccessful();
+    }
+
+    public void installCloudPluginInOpennmsMustBeSuccessful() throws Exception {
+        // Given
+        stack.opennms().copyFileToContainer(MountableFile.forHostPath(PATH_TO_FILE), format("%s%s", CONTAINER_PATH, KAR_FILE_NAME));
+        String config = createConfig();
+        await().atMost(20, SECONDS);
+
+        // When
+        KarafShellUtils.withKarafShell(stack.opennms().getSshAddress(), Duration.ofMinutes(3), streams -> {
+                streams.stdin.println(config);
+                streams.stdin.println("feature:install opennms-plugin-cloud-core");
+                streams.stdin.println("feature:list | grep opennms-plugin-cloud-core");
+                await().atMost(com.jayway.awaitility.Duration.ONE_MINUTE).until(() -> streams.stdout.getLines().stream().anyMatch(line -> line.contains("Started")));
+                streams.stdin.println("opennms-cloud:init key");
+                await().atMost(com.jayway.awaitility.Duration.ONE_MINUTE).until(() ->
+                        String.join("\n", streams.stdout.getLines()).contains("Initialization of cloud plugin in OPENNMS was successful"));
+                return true;
+        });
+    }
+
+
+    public void installCloudPluginInSentinelMustBeSuccessful() throws Exception {
+        // Given
+        stack.sentinel().copyFileToContainer(MountableFile.forHostPath(PATH_TO_FILE), format("%s%s", CONTAINER_PATH, KAR_FILE_NAME));
+        String config = createConfig();
+        await().atMost(20, SECONDS);
+
+        // When
+        KarafShellUtils.withKarafShell(stack.sentinel().getSshAddress(), Duration.ofMinutes(3), streams -> {
+            streams.stdin.printf("kar:install file:%s%s%n", CONTAINER_PATH, KAR_FILE_NAME);
+            streams.stdin.println("feature:install opennms-plugin-cloud-sentinel");
+            streams.stdin.println("feature:list | grep opennms-plugin-cloud-sentinel");
+            await().atMost(com.jayway.awaitility.Duration.ONE_MINUTE).until(() -> streams.stdout.getLines().stream().anyMatch(line -> line.contains("Started")));
+            streams.stdin.println(config);
+            try {
+                Thread.sleep(60000);
```

Comment:
Same here.

Reply:
This one not, otherwise I'm gonna get an exception ("command not found") by trying to init the opennms-cloud; I can of course try to reduce the sleep time

Reply:
You should `await` the command either by listing the commands (AFAIR `help`  should do this) or by just trying the command until it succeeds.